### ISSUE TITLE
extension: quick fix disable rule doesn't edit existing comment and creates duplicate disable comments instead

### ISSUE THREAD

**Issue by Miodec** [oxc-project/oxc#16060]
Version 1.30.0

If there are multiple linting errors on one line, using VSCodes Quick fix function, adds stacked disable comments instead of editing the existing comment, which means only the last disable comment is applied:

Expected result:
```ts
  // oxlint-disable-next-line no-unsafe-assignment no-unsafe-assignment strict-boolean-expressions
  window.dataLayer = window.dataLayer || [];
```

Actual result, when using the 'quick fix' option to disable the erroring rule:
```ts
  // oxlint-disable-next-line no-unsafe-assignment
  // oxlint-disable-next-line strict-boolean-expressions
  // oxlint-disable-next-line no-unsafe-assignment
  // oxlint-disable-next-line strict-boolean-expressions
  // oxlint-disable-next-line no-unsafe-assignment
  // ... and so on ...
  window.dataLayer = window.dataLayer || [];
```

Video:

https://github.com/user-attachments/assets/ccc1e830-8861-4668-9fcf-935730e587a2

Fix would be to either consider all stacked disable comments for the next valid line or edit the existing disable line if present.

**Comment by Sysix** [comment]
Relevant code snippet: https://github.com/oxc-project/oxc/blob/0069b042ca91f748be07a90135b3d50138855bee/crates/oxc_language_server/src/linter/error_with_position.rs#L220-L230

**Comment by overlookmotel** [comment]
Do we need to set a maximum line length after which goes onto new line?

```js
// oxlint-disable-next-line some-rule some-other-rule some-other-rule2 some-other-rule3 some-other-rule4 some-other-rule5 some-other-rule6 some-other-rule7 some-other-rule8 some-other-rule9
```

...is a bit silly.

Or maybe that's rare enough that we don't need to worry about it.

Edit: Discussed on Discord. Consensus (I think) is that we shouldn't worry about this. Apologies for the red herring!

**Comment by nrayburn-tech** [comment]
It’s an edge case for sure.

Does the linter support a multi-line disable comment? My assumption is no, but I’ve never tried it.
